scipy/stats/stats.py

```diff
@@ -8059,3 +8061,47 @@ def rankdata(a, method='average', *, axis=None):
 
     # average method
     return .5 * (count[dense] + count[dense - 1] + 1)
+
+
+class _StatsResult(dict):
+    """ Represents a statistics result. """
+    # stolen from OptimizeResult without modification
+    def __getattr__(self, name):
+        try:
+            return self[name]
+        except KeyError as e:
+            raise AttributeError(name) from e
+
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def __repr__(self):
+        if self.keys():
+            m = max(map(len, list(self.keys()))) + 1
+            return '\n'.join([k.rjust(m) + ': ' + repr(v)
+                              for k, v in sorted(self.items())])
+        else:
+            return self.__class__.__name__ + "()"
+
+    def __dir__(self):
+        return list(self.keys())
```

Comment:
@WarrenWeckesser Can/should this use `make_tuple_bunch` like `StatsTestResult = make_tuple_bunch('StatsTestResult', [], ['pvalue', 'statistic'])`, I read #3665 but didn't understand what should be done for new code.

Reply:
Also, generally `__repr__` is supposed to be something that could be `eval`'d to recreate the instance.  For a snippet like this

    result = pagel(data)
    print(repr(result))

I expect output of the form

    ClassName(method='exact', pvalue=0.0018191161948127822, statistic=133.5)

where `ClassName` is `StatsTestResult` if we stick with the generic result class, or the more specific `PageLResult` if we have a class specific to this test (either a root class or a subclass of `StatsTestResult`).

We could implement a separate `__str__` method if there is a friendlier output that we want to provide.

Reply:
This was taken straight from `OptimizeResult`. I'll take this out and just do what you did in `binomtest`.

Reply:
Just a reminder, @mdhaber started a discussion on representing the return values of the statistic functions over in gh-13118.  I'm testing some ideas and will comment there soon.  We might want to resolve that issue before digging into the details of how the return value of `pagel` (or any other new statistics function) is represented.